src/main/java/com/eppo/sdk/helpers/ConfigurationStore.java

```diff
@@ -71,12 +68,10 @@ public void setExperimentConfiguration(String key, ExperimentConfiguration exper
      *
      * @param key
      * @return
-     * @throws NetworkException
-     * @throws NetworkRequestNotAllowed
      * @throws ExperimentConfigurationNotFound
      */
     public ExperimentConfiguration getExperimentConfiguration(String key)
-            throws NetworkException, NetworkRequestNotAllowed, ExperimentConfigurationNotFound {
+            throws ExperimentConfigurationNotFound {
```

Comment:
My reading of this is that only `ExperimentConfigurationNotFound` is thrown

Reply:
🤷‍♂️ 
Maybe we should adjust the `catch` to be more restrictive

Reply:
you are suggesting replacing `Exception e` on line 77 with errors the `experimentConfigurationCache` is actually going to generate?

Reply:
Yeah, and letting other exceptions bubble up to our newly-added top-level try-catch